src/MainEmpresaProgramadores.java

```diff
@@ -1,103 +1,135 @@
-    empresa.addEmpleado(referencia);
-
-    referencia = new Programador(parametros);
-    empresa.addEmpleado(referencia);
-
-    referencia = new Gerente(parametros);
-    empresa.addEmpleado(referencia);
-
-    referencia = new Gerente(parametros);
-    empresa.addEmpleado(referencia);
-
-    referencia = new Programador(parametros);
-    empresa.addEmpleado(referencia);
-
-    referencia = new Programador(parametros);
-    empresa.addEmpleado(referencia);
-
-    referencia = new Programador(parametros);
-    empresa.addEmpleado(referencia);
-
-    referencia = new Programador(parametros);
-    empresa.addEmpleado(referencia);
+    Empresa empresa = new Empresa("Tecnologías Gyllenhaal");
+    int opcion;
+
+    Empleado referencia;
+
+    try {
+      referencia = new Programador("50549668S", "Daniel", 1302.93, "Java", true);
+      empresa.addEmpleado(referencia);
+    } catch (ParametroInvalidoException e) {
+      System.out.println(e.getMessage());
+    }
+
+    try {
+      referencia = new Programador("50782958C", "Max", 1342.55, "PL/SQL", true);
+      empresa.addEmpleado(referencia);
+    } catch (ParametroInvalidoException e) {
+      System.out.println(e.getMessage());
+    }
+
+    try {
+      referencia = new Gerente("50214985H", "Karla", 1824.73, "Contabilidad");
+      empresa.addEmpleado(referencia);
+    } catch (ParametroInvalidoException e) {
+      System.out.println(e.getMessage());
+    }
+
+    try {
+      referencia = new Gerente("02980444R", "Víctor", 1951.42, "Sistemas");
+      empresa.addEmpleado(referencia);
+    } catch (ParametroInvalidoException e) {
+      System.out.println(e.getMessage());
+    }
+
+    try {
+      referencia = new Programador("47274832J", "Valentín", 1300.00, "Java", true);
+      empresa.addEmpleado(referencia);
+    } catch (ParametroInvalidoException e) {
+      System.out.println(e.getMessage());
+    }
+
+
+    try {
+      referencia = new Programador("53688742H", "Fernando", 2159.02, "C++", false);
+      empresa.addEmpleado(referencia);
+    } catch (ParametroInvalidoException e) {
+      System.out.println(e.getMessage());
+    }
+
+    try {
+      referencia = new Programador("32831027M", "Michael", 1095.98, "Bash", true);
+      empresa.addEmpleado(referencia);
+    } catch (ParametroInvalidoException e) {
+      System.out.println(e.getMessage());
+    }
+
+    try {
+      referencia = new Gerente("07762415B", "Sergio", 2257.25, "Automatización");
+      empresa.addEmpleado(referencia);
+    } catch (ParametroInvalidoException e) {
+      System.out.println(e.getMessage());
+    }
 
     do {
       imprimirMenu();
-      opcion = scanner.nextInt();
-      scanner.nextLine(); // Limpiar buffer
+      opcion = Integer.parseInt(sc.nextLine());
 
       switch (opcion) {
-
-        case 1:
-          // Imprimir los datos de la empresa y todos sus empleados
-          empresa.toString();
-          break;
-
-        case 2:
+        case 1 ->
```

Comment:
Si compila con la versión de java 17 me vale, pero intentad no usar cosas en el examen que no hayamos visto en clase. Va a requerir más tiempo por mi parte de corrección y no voy a ir muy sobrado. Si tengo que buscar documentación sobre cosas que no hemos visto, necesito más tiempo, me voy cansando...

Si lo hacéis tal y como lo hemos visto en clase, lo marco como ok y paso a lo siguiente: yo soy más feliz y vosotras tenéis las notas antes. Gracias.

Reply:
Dependiendo del SDK que utilices el IDE autogenera o modifica ciertas cosas con la sintaxis que se usa en esa versión del JDK. Si dices que te vale, entonces en el examen es usable, no? Aunque entiendo que el switch lo vas a dar tú y que ya vendrá hecho por lo tanto no habría que modificar nada.

Reply:
Ante la duda, sigue la forma que hemos visto en los ejemplos de clase.